src/main/java/core/basesyntax/FileReaderImpl.java

```
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import core.interfaces.FileReader;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> read(String path) {
+        try {
+            return Files.readAllLines(Paths.get(path));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);

```

The catch block throws `UncheckedIOException(e)` which wraps the original exception but does not provide an informative message. This violates the checklist guidance: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an **informative** message and exception object." Provide a message that includes the `path`, e.g. `throw new RuntimeException("Can't read file by path: " + path, e);`. This makes failures easier to diagnose.